adobe/parliament-client-template
# [Feature] Swagger to graphql source plugin

## Description

Adds a new Gatsby source plugin that reads in a swagger file and produces GraphQL nodes for querying.

## Related Issue

None

## Motivation and Context

I could not find a swagger source plugin that provided the data we needed to properly create the pages we want.
This also gives us more control over how the GraphQL nodes are created and sourced.

## How Has This Been Tested?

Using a local instance already set up and pointed to an external doc project with a swagger file:

1. Install new dependencies: `yarn install`
1. Run tests: `yarn test`
1. Verify all Jest tests pass
1. Add a `SWAGGER_SOURCE_PATTERNS` environment variable to the development  and production `.env` files:

``` sh
SWAGGER_SOURCE_PATTERNS=**/petstore.yaml,**/petstore.json,**/swagger.json
```

5.  Run the development server
1. Verify development server successfully starts. (Some errors may be logged, but they are most likely not related to the changes in this PR).
1. Navigate to the graphQL playground
1. Verify the following types are listed on the left pane: `swaggerOpenApiInfo`,` swaggerOpenApiPath`, `swaggerOpenApiDefinition`
1. Create a graphql using these types
1. Verify data results reflects the data in the source files.
1. Stop the dev server
1. Run the build commands: `yarn build` or `yarn build:prod`
1. Verify static site is built successfully

## Screenshots (if appropriate):

## Types of changes




## Checklist:
